judge/ratings.py

```diff
@@ -122,20 +122,20 @@ def divconq(i, j):
             w_sum = 0.
             for j, h in enumerate(historical_p[i]):
                 gamma2 = (VAR_PER_CONTEST if j > 0 else 0)
-                h_var = get_var(times_ranked[i] + 1 - j, cache)
+                h_var = get_var(times_ranked[i] + 1 - j)
                 k = h_var / (h_var + gamma2)
                 w = w_prev * k**2
                 # Future optimization: If j is around 20, then w < 1e-3 and it is possible to break early.
-                tanh_terms.append((h, BETA2 ** .5 * TANH_C, w))
+                tanh_terms.append((h, math.sqrt(BETA2) * TANH_C, w))
                 w_prev = w
                 w_sum += w / BETA2
-            w0 = 1. / get_var(times_ranked[i] + 1, cache) - w_sum
+            w0 = 1. / get_var(times_ranked[i] + 1) - w_sum
             p0 = eval_tanhs(tanh_terms[1:], old_mean[i]) / w0 + old_mean[i]
             new_mean[i] = solve(tanh_terms, w0 * p0, lin_factor=w0)
 
     # Display a slightly lower rating to incentivize participation.
     # As times_ranked increases, new_rating converges to new_mean.
-    new_rating = [round(m - (get_var(t + 1, cache) ** .5 - VAR_LIM ** .5)) for m, t in zip(new_mean, times_ranked)]
+    new_rating = [round(m - (math.sqrt(get_var(t + 1)) - SD_LIM)) for m, t in zip(new_mean, times_ranked)]
```

Comment:
Should really have done `from math import sqrt`:

```console
$ python3 -m timeit -s 'a = 1337; from math import sqrt' 'sqrt(a)'
10000000 loops, best of 5: 26.6 nsec per loop
$ python3 -m timeit -s 'a = 1337; import math' 'math.sqrt(a)'
5000000 loops, best of 5: 46.1 nsec per loop
```

Reply:
Done. This saved 0.5 seconds, apparently. Rating all contests now takes 13 seconds.